### PR TITLE
Fix geohash string type

### PR DESCRIPTION
**Pull request by mRoca**
This PR fixes the `Geohash` class for PHP 8.1 :

```
 strlen(): Passing null to parameter #1 ($string) of type string is deprecated in vendor/league/geotools/src/Geohash/Geohash.php line 140
``` 

If anyone needs to use the Geohash class with PHP 8.1, you can create this one while waiting for this PR to be merged:

```php
use League\Geotools\Geohash\Geohash as Base;

class Geohash extends Base
{
    protected $geohash = '';
}

```

### CODE REVIEW

**Comment by Surfoo** [comment]
Thank you, I just fixed a unit test, could you rebase please?

**Comment by mRoca** [comment]
> Thank you, I just fixed a unit test, could you rebase please?

done 👍🏻

File: src/Geohash/GeohashInterface.php [hunk 1]
```diff
@@ -24,13 +26,13 @@ interface GeohashInterface
...
      *
      * @param CoordinateInterface $coordinate The coordinate to encode.
-     * @param integer             $length     The length of the hash between 1 to 12 by default (optional).
+     * @param int                 $length     The length of the hash between 1 and 12 by default (optional).
      *
      * @return GeohashInterface
      *
      * @throws InvalidArgumentException
      */
-    public function encode(CoordinateInterface $coordinate, $length = self::MAX_LENGTH);
```
Comment (Surfoo) [inline src/Geohash/GeohashInterface.php:33]: Why did you remove the default value of `$length`? It's a breaking change for the GeohashInterface.

Comment (mRoca) [reply src/Geohash/GeohashInterface.php:33]: This file refers to an Interface, it's not the class itself. Therefore, `self::MAX_LENGTH` cannot be resolve at this place, as `self::` refers to nothing, and `public const MAX_LENGTH = 12;` is only defined in the Geohash class, not in the Interface

Comment (mRoca) [reply src/Geohash/GeohashInterface.php:33]: The default value is still present in the `Geohash::encore` method definition

Comment (Surfoo) [reply src/Geohash/GeohashInterface.php:33]: Exact, I didn't notice the default value in `Geohash::encode`. You could remove `(optional)` in the comment above.
